flowforge/input/Components.py

```diff
@@ -423,52 +551,57 @@ def getOutlet(self, inlet: Tuple[float, float, float]) -> Tuple[float, float, fl
 
     def getVTKMesh(self, inlet: Tuple[float, float, float]) -> VTKMesh:
-        return genUniformCylinder(self._L, self._R, naxial_layers=self._n, **self._kwargs).translate(
-            inlet[0], inlet[1], inlet[2], self._theta, self._alpha
-        )
+        # Fix to prevent R from being passed twice in case a circular cross section is used
+        # TODO: change to return different mesh shapes when using different cross section types
+        filtered_kwargs = {k: v for k,v in self._kwargs.items() if k!= 'R'}
+        return genUniformCylinder(self._L, self._Dh/2, naxial_layers=self._n, **filtered_kwargs).translate(
+            inlet[0], inlet[1], inlet[2], self._theta, self._alpha)
 
     def getBoundingBox(
         self, inlet: Tuple[float, float, float]
     ) -> Tuple[Tuple[float, float, float], Tuple[float, float, float], float, float, float, float, float]:
         outlet = self.getOutlet(inlet)
-        return [inlet, outlet, self._R, self._R, self._L, self._theta, self._alpha]
+        return [inlet, outlet, self._Dh/2, self._Dh/2, self._L, self._theta, self._alpha]
 
     def _convertUnits(self, uc: UnitConverter) -> None:
         self._L *= uc.lengthConversion
-        self._R *= uc.lengthConversion
         self._Ac *= uc.areaConversion
         self._Dh *= uc.lengthConversion
         self._Pw *= uc.lengthConversion
         self._roughness *= uc.lengthConversion
         self._heatedPerimeter *= uc.lengthConversion
 
-
-component_list["pipe"] = Pipe
-
-
-class SquarePipe(Pipe):
-    """A square pipe component
-
-    Parameters
-    ----------
-    L : float
-        Length of the pipe
-    W : float
-        Width of the pipe (i.e. flat-to-flat)
-    """
-
-    def __init__(self, L: float, W: float, **kwargs) -> None:
-        super().__init__(L=L, Dh=W, Ac=W**2, **kwargs)
-
-    def getVTKMesh(self, inlet: Tuple[float, float, float]) -> VTKMesh:
-        return genUniformCube(self._Dh, self._Dh, self._L, nz=self._n).translate(
-            inlet[0], inlet[1], inlet[2], self._theta, self._alpha
-        )
-
-
-component_list["square_pipe"] = SquarePipe
-
-
+def create_pipe(L, *args,
+                cross_section_name="circular",
+                n=1,
+                theta=0.0,
+                alpha=0.0,
+                Klossinlet=0.0,
+                Klossoutlet=0.0,
+                Klossavg=0.0,
+                roughness=0.0,
+                pctHeated=1.0,
+                **kwargs) -> Pipe:
+    return Pipe(
+        L=L,
+        cross_section=cross_section_classes[cross_section_name](
+            **{k: v for k, v in kwargs.items()
+               if k in cross_section_param_lists[cross_section_name]}
+        ),
+        n=n,
+        theta=theta,
+        alpha=alpha,
+        Klossinlet=Klossinlet,
+        Klossoutlet=Klossoutlet,
+        Klossavg=Klossavg,
+        roughness=roughness,
+        pctHeated=pctHeated,
+        *args,
+        **kwargs
+    )
+component_list["pipe"] = create_pipe
```

Comment:
Actually..... Looking at this more closely, why didn't you just go with `component_list["pipe"] = Pipe`?